### ISSUE TITLE
[P2] [Team 5] [Phase 2] Discard Activates For Players With Fewer Than 7 Resources

### ISSUE THREAD

**Issue by dannya951** [cs340tabyu/cs340Winter2017#359]
When a 7 is rolled, as long as at least one person needs to discard, then all players are required to discard.  If one of the players has 0 resources, then they are never able to finish discarding because they never press an up arrow, which is where the discard button gets disabled.  Thus, they get stuck in the discard modal and the game can't proceed.

**Comment by xandroxygen** [comment]
Unable to reproduce at all, marking invalid.

Game 0 with Sam Brooke Pete and Mark:
Mark's turn, he rolled a 7.
Sam needs to discard because she has > 7 cards:
![image](https://cloud.githubusercontent.com/assets/20050280/19774791/7e01658e-9c2b-11e6-8110-ed8a5c3fbad4.png)

Brooke has exactly 7 cards and wasn't asked to discard:
![image](https://cloud.githubusercontent.com/assets/20050280/19774862/abc09e4a-9c2b-11e6-908f-50024017ebd0.png)

Pete also had < 7 cards and wasn't asked to discard:
![image](https://cloud.githubusercontent.com/assets/20050280/19774876/c6799e6c-9c2b-11e6-8110-c8c32448a3ce.png)

And Mark had < 7 and wasn't asked to discard:
![image](https://cloud.githubusercontent.com/assets/20050280/19774912/e53c7928-9c2b-11e6-911f-9d1a7fa5abb2.png)

Are you sure you were testing with Team 5's code? because the spreadsheet is confusing, one of our team members accidentally tested our own code yesterday.

**Comment by dannya951** [comment]
Was your code not the code in D10?

**Comment by xandroxygen** [comment]
Yep it was, good to know you got the right code! still unable to reproduce.

**Comment by dannya951** [comment]
I just re-downloaded everything.  Let me try again.

**Comment by dannya951** [comment]
<img width="960" alt="discard_error" src="https://cloud.githubusercontent.com/assets/21351700/19775599/302cbd92-9c2e-11e6-932c-c6ed452a415f.png">

**Comment by dannya951** [comment]
This is using the client I just downloaded in 5.zip (cell D10 on sheet) and the demo server.

**Comment by dannya951** [comment]
This is a good example of the P2 case, also, since the bottom left player will never be able to discard.

**Comment by dannya951** [comment]
Nevermind, I was able to discard on bottom left, but only after offering 0, then 1, then 0, then -1, then 0 (thus criterion is met upon clicking up arrow).  Really weird, haha.  I actually hadn't seen that occur in testing.  But ya, if the player has 0 resources, it blocks the game.

**Comment by xandroxygen** [comment]
So I wasn't able to reproduce and you were.... the only difference I can spot is that you had 4 clients open at once, and I logged in and out only using 1 client. (also that discard thing is weird haha, I don't think our client supports offering -1). Can you reproduce with only 1 client open, or 4 separate clients on different computers? (if your team isn't free right now to do that, no worries)

**Comment by dannya951** [comment]
I'll do that later and try to get back to you by the end of the day.

**Comment by dannya951** [comment]
I still haven't tried single client play, but here are some screenshots of the negative resources in the discard.
<img width="481" alt="1" src="https://cloud.githubusercontent.com/assets/21351700/19818728/5296077c-9d0e-11e6-8d77-60d8ad3967ec.png">
<img width="481" alt="2" src="https://cloud.githubusercontent.com/assets/21351700/19818730/54009938-9d0e-11e6-84da-180ec149eb4c.png">
<img width="480" alt="3" src="https://cloud.githubusercontent.com/assets/21351700/19818731/5567def8-9d0e-11e6-90f3-68c9722e7d70.png">
<img width="483" alt="4" src="https://cloud.githubusercontent.com/assets/21351700/19818733/56683988-9d0e-11e6-9a9f-a3afbbe378d2.png">
